locks/dynamodb/dynamo_lock.go

```diff
@@ -86,8 +87,19 @@ func (dynamoDbLock DynamoDbLock) ReleaseLock(terragruntOptions *options.Terragru
 		return err
 	}
 
-	if err := removeItemFromLockTable(dynamoDbLock.StateFileId, dynamoDbLock.TableName, client); err != nil {
-		return err
+	err = removeItemFromLockTable(dynamoDbLock.StateFileId, dynamoDbLock.TableName, client)
+
+	// If the table or lock doesn't exist log it, else return the error
+	if err != nil {
+		if awsErr, isAwsErr := errors.Unwrap(err).(awserr.Error); isAwsErr && awsErr.Code() == "ResourceNotFoundException" {
+			terragruntOptions.Logger.Printf("Table %s does not exist in DynamoDB!", dynamoDbLock.TableName)
+			return nil
+		} else if awsErr, isAwsErr := errors.Unwrap(err).(awserr.Error); isAwsErr && awsErr.Code() == "ConditionalCheckFailedException" {
+			terragruntOptions.Logger.Printf("Nothing to release, specified lock does not exist in table %s", dynamoDbLock.TableName)
+			return nil
+		} else {
+			return err
+		}
```

Comment:
Also, if you do return the custom error type, you could move this code into the `removeItemFromLockTable` code itself so all callers benefit from better error handling.